product_docs/docs/pgd/5/admin-tpa/upgrading_major_rolling.mdx

```diff
@@ -0,0 +1,537 @@
+------------------------------------------------------
+Collecting old cluster control data                         ok
+Collecting new cluster control data                         ok
+Checking LSN of new cluster                                 ok
+Starting new cluster with BDR disabled                      ok
+Connecting to new cluster                                   ok
+Creating replication origin (bdr_bdrdb_rb69_bdr2)           ok
+Advancing replication origin (bdr_bdrdb_rb69_bdr2, 0/1F4... ok
+Creating replication origin (bdr_bdrdb_rb69_bdr1)           ok
+Advancing replication origin (bdr_bdrdb_rb69_bdr1, 0/1E8... ok
+Creating replication slot (bdr_bdrdb_rb69_bdr1)             ok
+Creating replication slot (bdr_bdrdb_rb69)                  ok
+Creating replication slot (bdr_bdrdb_rb69_bdr2)             ok
+Stopping new cluster
+```
+
+### Modify the Postgres service
+
+The Postgres service on the system is configured to start the old version 
+of Postgres (PG15). You need to modify the `postgres.service` file to start the new 
+version (PG16). You can do this using `sed` to replace the old version
+number `15` with `16` throughout the file. 
+
+After you've changed the version number, you can tell the 
+systemd daemon to reload the configuration. On the target node, run:
+
+```bash
+sudo sed -i -e  's/15/16/g' /etc/systemd/system/postgres.service
+sudo systemctl daemon-reload
+```
+
+### Start Postgres
+
+Start the modified Postgres service by running:
+
+```bash
+sudo systemctl start postgres
+```
+
+### Validate the Postgres version
+
+Repeating the first step, check the version of Postgres to confirm that you upgraded kaboom correctly. While still on kaboom, run:
+
+```bash
+sudo -u postgres psql bdrdb
+```
+
+```sql
+SELECT * FROM bdr.group_versions_details;
+```
+
+The output is similar to this for your cluster. Use the output to confirm that kaboom is running the upgraded Postgres version. 
+
+```
+  node_id   | node_name |        postgres_version        | bdr_version 
+------------+-----------+--------------------------------+-------------
+ 3490219809 | kaftan    | 15.5 (Debian 15.5-1EDB.buster) | 5.3.0
+ 2710197610 | kaboom    | 16.1 (Debian 16.1-1EDB.buster) | 5.3.0
+ 2111777360 | kaolin    | 15.5 (Debian 15.5-1EDB.buster) | 5.3.0
+(3 rows)
+```
+
+### Test the upgrade
+
+After the upgrade is complete on kaboom, test your databases thoroughly to ensure the upgrade was successful. Being still connected to kaboom ensures that your tests run specifically against the new version without interference from cluster routing mechanisms.
+
+```bash
+sudo -u postgres psql <database_name>
+```
+
+Next, execute your test suite, including application transactions, queries, and other operations critical to your environment. Pay particular attention to performance metrics and compatibility with the new version. The insights gained from this controlled upgrade can inform your approach, helping to ensure a smoother transition for all nodes.
```

Comment:
Well thats all well and good but did we mention "You should have a test suite you can deploy on any node" in the preamble? 

Also isn't the node returning to the cluster at this point?

Have we checked it is/isn't returning?

Reply:
Yeah, I will add something about needing a test suite in the preamble. 

For your second/third point here, did we ever remove the node from the cluster in the first place? I don't think I documented that we need to remove the node only that we need to stop postgres on the node. I assumed PGD would just not use it with postgres turned off on the node and that it would use it again with postgres turned back on on the node. What do I need to do to remove and or rejoin a node? Rejoining presumably only happens after the test suite has been run, correct?

Reply:
Second/third point is about running a test suite on a node thats in the process of resyncing with the cluster. I don't recall this step in previous guides so not sure where this came from, but it feels potentially dangerous as a step.

Reply:
It's only dangerous if the node is syncing back with its group at the time of the testing right? The step to test seems right as part of upgrading process, correct?

Reply:
I am pulling the test suite content.